src/lib.rs

```diff
@@ -117,16 +117,30 @@ impl BarcodeScanner {
 		loop {
 			let events = self.device.fetch_events()
 				.map_err(|e| Error::new(format!("Failed to fetch events from input device: {e}")))?;
+
+			// Track the state of the shift keys and capslock
+			let mut left_shift_pressed = false;
+			let mut right_shift_pressed = false;
+			let mut capslock_on = false;
 			for event in events {
 				// Check if key is pressed (value 1 for the key pressed, velue 0 for the key released).
-				if event.event_type() == evdev::EventType::KEY && event.value() == 1 {
+				if event.event_type() == evdev::EventType::KEY {
 					// Create Key object based on the code.
 					let key_name = evdev::Key(event.code());
 
-					// Map key_name to the number or char.
-					if let Some(c) = key_to_str(key_name) {
-						self.buffer.push(c);
-					}
+					match key_name {
+						evdev::Key::KEY_LEFTSHIFT => left_shift_pressed = event.value() == 1,
+						evdev::Key::KEY_RIGHTSHIFT => right_shift_pressed = event.value() == 1,
+						evdev::Key::KEY_CAPSLOCK => capslock_on = event.value() == 1,
+						_ => {},
+					}					
+
+                    // Map key_name to the number or char.
+                    if event.value() == 1 {
+                        if let Some(c) = key_to_str(key_name, left_shift_pressed || right_shift_pressed || capslock_on) {
```

Comment:
This isn't totally correct: capslock only shifts letters to their uppercase variant. It doesn't shift numbers or symbols.